### ISSUE TITLE
taxonomy of ARF report ?

### ISSUE THREAD

**Issue by isimluk** [OpenSCAP/foreman_openscap#38]
See  c911d65fc965a45c142e8a735ba1c698b030e255 and ed8e64d97081edd25c0d1e3909acdcbe97f08a3a.

Me and @ares came to the conclusion that ArfReport should always be in the same org/location as the Host is. Hence, we perhaps don't need to have separate taxonomy. We just need to create a default_scope for ArfReport that would reflect current org/loc.

Thoughts?

**Comment by shlomizadok** [comment]
I think it is okay. Anyhow, if you assign policy to a host it would be in the context of the taxonomies.
Added a default_scope to ArfReport in #39

**Comment by isimluk** [comment]
The problem is when a host is migrated to different taxonomy, then the /compliance/reports break, because the previous organization will not see the report's host (the host would be nil I think).

I'll leave this question open for now. But perhaps it is not a top priority.

**Comment by shlomizadok** [comment]
This was actually fixed in #39. Can you re-test?
For safety, maybe we can refactor to ```<%= name_column(arf_report.host) if arf_report.host %>```

**Comment by isimluk** [comment]
Ok, will retest. Thanks!

**Comment by isimluk** [comment]
Right. I was not able to break /compliance/reports page. I was not able to change the org/loc of existing host.

**Comment by ares** [comment]
What about scap content taxonomies? If we remove taxonomies from policy we should do the same for scap contents but here I'm not sure whether it makes sense. In multitenancy setup I see use case where you want scoped scap content. And then there's a question whether policy should "inherit" taxonomies from content or we should even allow subset of content taxonomies.

If we don't have easy answers here I'm thinking of leaving taxonomies as it is now. Warning: I'm still not confident with all details of openscap plugin so I may be completely wrong.

**Comment by shlomizadok** [comment]
I am in favor of making scap_content system-wide and not bind scap_content
to taxonomies. I think it should be system-wide, because you can set only
*one* scap_content from a file (e.g., you can't create 'fedora'
scap_content twice - one for org x and one for org y). Therefore, I think
that scap_content should be without taxonomies at all.

2015-01-08 13:44 GMT+02:00 Marek Hulán <notifications@github.com>:

**Comment by isimluk** [comment]
@shlomizadok , I agree with @ares that in multi-tenant environment: there might contents that should not be shared.

@ares it is interesting idea to inherit taxonomy from content to the policy and keep it the same.

But I have started this thread because something different. I suggested to drop taxonomy from ArfReport, because the ArfReport belongs to host and makes no sense to share outside of the host's org/loc.

That is perhaps doable by default scope on ArfReport.
```ruby
default_scope {
    joins(:host)
}
```
But I am not sure. I have yet to dive to the Foreman's taxonomies a bit more.

**Comment by ares** [comment]
Ah sorry, I mixed things. So in terms of ArfReport, it seems that we should drop taxonomy since it should always be determined by host. In my opinion default_scope is never good idea but in general, you're right, we should be able to define relation to taxonomy like `belongs_to :organization, :through => :host` a vice versa. But at the same time I think you don't even need to mess with taxonomies, you'll always display report for specific host or set of hosts. (Unless you have some generic dashboard that's completely unrelated to hosts?)

**Comment by isimluk** [comment]
Ah, ok. So the default_scope will not be that easy.

**Comment by shlomizadok** [comment]
So we've already established that arf_report's taxonomy is the host's taxonomy. 
Can we close this issue?
